cmd/api/src/database/helper.go

```diff
@@ -17,16 +17,24 @@
 package database
 
 import (
+	"errors"
+	"fmt"
 	"github.com/gofrs/uuid"
-	"github.com/specterops/bloodhound/errors"
 	"gorm.io/gorm"
+	"strings"
 )
 
 func CheckError(tx *gorm.DB) error {
 	if errors.Is(tx.Error, gorm.ErrRecordNotFound) {
 		return ErrNotFound
 	}
 
+	if tx.Error != nil {
+		if strings.Contains(tx.Error.Error(), "duplicate key value violates unique constraint \"users_principal_name_key\"") {
```

Comment:
We want to keep this function agnostic to any specific db table as it's used across the datalayer. This error handling might be better placed in the db method directly associated with user creation

Reply:
Yea in both is likely a better place than here IMO